### DISCUSSION TITLE
No data from EPEVER

### DISCUSSION THREAD

**Discussion by kalle3510** [softwarecrash/EPEver2MQTT#56] [Q&A]
Does the EPEVER MPPT XTRA4415N also work with your solution? 
Errorcode 2 time flash, no data from EPEVER.
When I look with the oszilloscope, data reaches Wemos.
Unfortunately I am not familiar, how the data should look like. :-(

**Comment by softwarecrash** [comment]
Have you checked the epever have ID 1?

**Comment by kalle3510** [comment]
Yes, ID is 1Am 03.11.2024 um 11:25 schrieb Rocky ***@***.***>:﻿
Have you checked the epever have ID 1?

**Reply by all-solutions** [reply]
Can you please provide fotos of your wiring?

**Reply by kalle3510** [reply]
Of course, thanks for your request.
![IMG_4921](https://github.com/user-attachments/assets/aed564b9-5760-42fd-89e3-d00090f5e420)
![IMG_4922](https://github.com/user-attachments/assets/2dc7dbac-2896-4188-a76a-2f9dc41326ad)
![IMG_4924](https://github.com/user-attachments/assets/abfb0551-4e2a-482f-82a9-aa278cd01fe9)
![IMG_4926](https://github.com/user-attachments/assets/3d75e1d9-ae0d-4b93-86de-9e8634d2db89)
![IMG_4927](https://github.com/user-attachments/assets/05d25f34-7978-4f59-808f-b045bc925ce4)
![IMG_4928](https://github.com/user-attachments/assets/db4b8834-07bb-401a-859b-8b5fd6aea9eb)
![IMG_4929](https://github.com/user-attachments/assets/7fc88d01-df71-4d6f-addd-e0d14779e023)
![IMG_4930](https://github.com/user-attachments/assets/9635a2c3-0ba6-44f2-b2b6-6777a58a66a9)
![IMG_4931](https://github.com/user-attachments/assets/89d1b2fc-7337-4746-a3b0-b6f1ff711533)

**Reply by softwarecrash** [reply]
thank you, can you post the esp-ip/livejson data please

**Reply by kalle3510** [reply]
I am at the end of my knowledge?  
Perhaps you may help me how to post the esp-ip/livejson data. 
You are not talking about the WebSerial information, see picture.
![Serial info](https://github.com/user-attachments/assets/4ec6ff31-f714-44f2-afd0-18a20f531490)

**Reply by softwarecrash** [reply]
Just Type in your Browser http://your.esp.ip/livejson

**Comment by kalle3510** [comment]
Thats all?

{"DEVICE_NAME":"EPEver2MQTT"}

 

 

Von: Rocky ***@***.***> 
Gesendet: Sonntag, 3. November 2024 18:07
An: softwarecrash/EPEver2MQTT ***@***.***>
Cc: kalle3510 ***@***.***>; Author ***@***.***>
Betreff: Re: [softwarecrash/EPEver2MQTT] No data from EPEVER (Discussion #56)

 

Just Type in your Browser http://your.esp.ip/livejson

**Comment by kalle3510** [comment]
I did, and that was all what is shown on the screen
{"DEVICE_NAME":"EPEver2MQTT"}

**Reply by softwarecrash** [reply]
Thats Not much. Maybe, some epever havent enugh Power to Drive the ESP and RS Board, have you checked stable suply or tried with external Power?

**Reply by kalle3510** [reply]
not now, I also read this in the other comments. I will check and come back. (Datasheet tells me 150mA for the interface.) Thank you for your support so fare.

**Reply by kalle3510** [reply]
Today I added a 850mA 5V power supply and removed the two supply wires from EPEVER. Unfortunately no change.

**Reply by softwarecrash** [reply]
ah i think i found it, the epever have a bit different modbus or register adresses. we need the actual datasheet for it

**Reply by kalle3510** [reply]
is this part of the user manual?
[EPEVER Laderegler XTRA4415N-SMS-EL-V1.3.pdf](https://github.com/user-attachments/files/17622450/EPEVER.Laderegler.XTRA4415N-SMS-EL-V1.3.pdf)

**Reply by softwarecrash** [reply]
no, it looks like this 
https://github.com/softwarecrash/EPEver2MQTT/blob/main/Protocol/ControllerProtocolV2.3.pdf

**Reply by kalle3510** [reply]
I will contact the epever support

**Comment by kalle3510** [comment]
I found this one online: [https://github.com/B48D81EFCC/esphome-epever-xtra-n-series](url)

**Comment by kalle3510** [comment]
I received an answer from EPEVER, does this help?
[A or BSeriesControllerProtocolv2.6.pdf](https://github.com/user-attachments/files/17635564/A.or.BSeriesControllerProtocolv2.6.pdf)

**Comment by softwarecrash** [comment]
yes, but this is a known old protocol sheet.
let me check later that i make you a version without the static request and added some debug

**Comment by kalle3510** [comment]
Old protocol, ok. 

That would be great, thank you in advance.

 

Von: Rocky ***@***.***> 
Gesendet: Dienstag, 5. November 2024 17:34
An: softwarecrash/EPEver2MQTT ***@***.***>
Cc: kalle3510 ***@***.***>; Author ***@***.***>
Betreff: Re: [softwarecrash/EPEver2MQTT] No data from EPEVER (Discussion #56)

 

yes, but this is a known old protocol sheet.
let me check later that i make you a version without the static request and added some debug

**Comment by agaletski** [comment]
Hello.
I had the similar issue. I my case all hardware is the same except I use ESP01 instead of D1 mini. My MPPT is XTRA4415N-XDS2.
The issue was resolved after checked and corrected wiring between MAX485 and ESP modules. 
DE an RE leads used to enable read/write should be connected to Gpio0 on ESP module.  
I have checked provided photos wiring , they looks correct according to wiring provided in Wiki, however, when I checked D1 mini board pinout reference it become clear that Wiki may provide wrong scheme for D1 mini board connection.
As mentioned above DE & RE should be connected to Gpio0 and it works for me (ESP01 module)
For D1 mini board Gpio0 corresponds to pin labeled D3. In Wiki we use D1 which is Gpio5. 
I see there are different binaries provided for  ESP01 and D1 mini boards and I guess differences are in GPIO assignment.
Possible mistake could also happen during GPIO configuration before compile to binary. 

Great thanks for this project. I works fine for my  XTRA4415N-XDS2.

**Reply by kalle3510** [reply]
Hi, thank you so much for your point. I will check and come back.

**Reply by kalle3510** [reply]
Hi again, I checked my setup.  DE/ RE signal is visible on D1 not on D3 unfortunately. :-(

**Reply by softwarecrash** [reply]
Follow.the wireing Guide, binarys have different assigment for each ESP Model.
When it Doenst Work, Check Power supply and Set device to ID1 or replace the max485

**Reply by kalle3510** [reply]
a)	Follow the wireing Guide    =>done, was also checked before

b)	binarys have different assigment for each ESP Model. => used the one suitable form y D1

c)	When it Doenst Work, Check Power supply => also tried external supply

d)	and Set device to ID1	=> checked again, I did

e)	or replace the max485	=> was done, no change

	I measured the following signals on 
Channel 1 TX; Channel 2 D1
 
![image](https://github.com/user-attachments/assets/29d3e91d-c420-440f-965a-6d8f55bc32a5)


Channel 1 RX; Channel 2 D1
 
![image](https://github.com/user-attachments/assets/f81fcbc4-dc0b-43d2-bae2-68f15ec9bbbc)



Why ist the receive signal RX not synced with D1?

**Reply by softwarecrash** [reply]
the image for ESP01 have GPIO0 for DE-RE reason is limited pinout
the image for Wemos D1 Mini (and most ohter similar PCB) have have GPIO5 for DE-RE

the signal measurement is from wich side? rs485 or the ttl side?
the measured signal looks like the blue have a short to yelow, so when it send something the yelow will dropped

what say the webserial console?

sometimes users reported a defective rs485 port on the device, have you checked it with pc connection or something else?

**Reply by kalle3510** [reply]
I am using D1 Mini, so GPIO5 is correct

![image](https://github.com/user-attachments/assets/560e3579-a40d-458f-bed8-ead9b9fd8af2)
signal measaurement is at the D1 Mini=> D1 RX TX

blue short to yellow, I also do not understand the voltage drop, but VDD is stable 
what about the RX signal (yellow) before DE/RE signal (blue)

The consol tells me no connection.
Also 2 time flashing of the LED.

Yes the EPEVER is working fine. I am using a EPEVER Wifi Adapter 2.4G RJ45 D, but I want to replace it by your device.

**Reply by softwarecrash** [reply]
What say the webserial? I remember i make it Output the reason Code when fail

The rs485 IS powered over what? 5v or 3v3?

**Comment by kalle3510** [comment]
WEB serieal:   [1] 2 Read Statistical Data Failed

I have tested both: 485 powere from EPEVer with 5V, or from additional supply over WEMO with 5V

**Reply by softwarecrash** [reply]
well that helps now!
there is a communication issue in the statistic data section, its super fuzzy to find whats exactly wrong without the device that answer with this "wrong" data
let me check it.

what for a device you have?
can you join the discord channel? it needs a bunch of firmwares to find out what data is wrong

**Reply by kalle3510** [reply]
EPEVER MPPT XTRA4415N

discord channel: what do Ihave to do

**Reply by softwarecrash** [reply]
> EPEVER MPPT XTRA4415N
> 
> discord channel: what do Ihave to do

just follow the link https://discord.gg/Hup3gg4YsN

**Reply by kalle3510** [reply]
done

**Reply by softwarecrash** [reply]
[EPEver2MQTT_esp01_1m_2.3.2B1.zip](https://github.com/user-attachments/files/18274678/EPEver2MQTT_esp01_1m_2.3.2B1.zip)
please try this  and  check the webserial for output

**Comment by kalle3510** [comment]
I flashed the new bin-file.
Webconsol: 226 read registers for clock failed

**Reply by softwarecrash** [reply]
226 means a communication timeout, so at the moment No answer from your device or the requests was Not arrived

**Comment by kalle3510** [comment]
Please have a look to the post before, where I added the oscillogramm for RX (xellow).
Is it right, that RE&DE on Pin D1 (blue) is later than Rx ?

**Reply by softwarecrash** [reply]
Yes, re-de is enabled bevor sending Data (ESP Side)
So after rx Signal the ESP enable IT and sending the next request

**Comment by MaxXx44** [comment]
Hello,
Have you found a solution, I also have an epever xtra 4415 and I have the same problem as you

**Reply by softwarecrash** [reply]
the xtra 4415 are confirmed, i think your setup is wrong

**Reply by MaxXx44** [reply]
Thank you for your answer
I also have an epever tracer An, when I plug it in it works very well
If I plug my tracer An module into the Xtra it doesn't work either
No data
first module ID 1 and second ID 2

**Reply by MaxXx44** [reply]
Modele: Xtra4415N-G3

**Reply by softwarecrash** [reply]
> Thank you for your answer I also have an epever tracer An, when I plug it in it works very well If I plug my tracer An module into the Xtra it doesn't work either No data first module ID 1 and second ID 2

that doenst  work in this way, in single mode  the ID must be 1. when you set up more devices, it must all connected and enumerated

**Reply by MaxXx44** [reply]
My 2 epever are not connected together on the same battery, they are independent
2 different systems (they are not connected to each other)
I tried to disconnect the first when I connect the second it does not change anything
I will redo everything while keeping id 1, disconnect the second and I will keep you informed

**Reply by softwarecrash** [reply]
> My 2 epever are not connected together on the same battery, they are independent 2 different systems (they are not connected to each other) I tried to disconnect the first when I connect the second it does not change anything I will redo everything while keeping id 1, disconnect the second and I will keep you informed

when you swap your esp from one to another epever, the Device ID must allways 1.
the program aks slave 1 for data, when your second epever have ID 2 it will never asked for data

**Reply by MaxXx44** [reply]
My 2 esp work on the epever tracer An, and the 2 esp do not work on the epever Xtra G3
I took a new vemos D1 and I redid everything from scratch leaving id at 1
Unfortunately it still does not want to work on the Xtra G3 but it works on the tracer

**Comment by MaxXx44** [comment]
Mine works on epever tracer but does not work on an Xtra

**Comment by all-solutions** [comment]
Question: You have two epevers, even if they are completely separate. Are the two connected to the SAME Wemos?

**Reply by MaxXx44** [reply]
No 2 different Wemos on 2 independent epever
the 2 wemos D1 works on the epever tracer An but does not work on the epever Xtra 4415N G3

**Comment by MaxXx44** [comment]
led lights flash 2 times every 5 seconds
normally no connection

**Comment by MarcoHacs** [comment]
@MaxXx44  @kalle3510 
Did you found a Solution ?

**Comment by kalle3510** [comment]
No, unfortunately not.

 

Von: MarcoHacs ***@***.***> 
Gesendet: Samstag, 31. Mai 2025 02:16
An: softwarecrash/EPEver2MQTT ***@***.***>
Cc: kalle3510 ***@***.***>; Mention ***@***.***>
Betreff: Re: [softwarecrash/EPEver2MQTT] No data from EPEVER (Discussion #56)

 

@MaxXx44 <https://github.com/MaxXx44>  @kalle3510 <https://github.com/kalle3510> 
Did you found a Solution ?

**Comment by softwarecrash** [comment]
G3 devices now support

**Comment by kalle3510** [comment]
Great job, Have just seen your message. Works as expected. Thank you very much!
